### PR TITLE
ARC-1177 deploy fixed version of Github client replacement of Octokit

### PR DESCRIPTION
**Pull request by joshkay10**
This is a re pull request of the failed deployment.
- Cause was that the commits Octokit call was missing an await. Have highlighted it with a comment,
- I have also added Testing so that the Octokit variation is tested, also commented to highlight.

### CODE REVIEW

File: src/sync/commits.ts [hunk 1]
```diff
@@ -30,37 +29,22 @@ export const getCommits = async (logger: LoggerWithTarget, github: GitHubAPI, gi
 			jiraPayload
 		};
 	}
-	console.log("IM GETTING FROM OCTO");
-	return getCommitsOctoKit(logger, github, gitHubClient, jiraHost, repository, cursor, perPage) 
+	return await getCommitsOctoKit(logger, github, gitHubClient, jiraHost, repository, cursor, perPage);
```
Comment (joshkay10) [inline src/sync/commits.ts:32]: This was the await that was missing

File: test/safe/branch.test.ts [hunk 2]
```diff
@@ -38,9 +38,15 @@ describe("Branch Webhook", () => {
 		await sqsQueues.branch.purgeQueue();
 	});
 
-	describe("Create Branch", () => {
+	describe.each([true, false])("Create Branch - New GH Client feature flag is '%s'", (useNewGithubClient) => {
```
Comment (joshkay10) [inline test/safe/branch.test.ts:41]: Made sure both Octokit and Github client variation get tested, catch it before deployment!!

File: test/unit/sync/commits.test.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ const commitsNoKeys = require("../../fixtures/api/graphql/commit-nodes-no-keys.j
 jest.mock("../../../src/sqs/queues");
 jest.mock("../../../src/config/feature-flags");
 
-describe("sync/commits", () => {
+describe.each([true, false])("sync/commits - new github client is %s", (useNewGithubClient) => {
```
Comment (joshkay10) [inline test/unit/sync/commits.test.ts:23]: Made sure both Octokit and github client are tested for the commits flow.

**Review by KAbakumovAtl — APPROVED** [review]

File: src/github/branch.ts [hunk 4]
```diff
@@ -29,19 +26,19 @@ export const createBranch = async (
...
 
-		const jiraPayload = await transformBranch(context.github, webhookPayload);
+		const gitHubClient = new GitHubClient(getCloudInstallationId(githubInstallationId), context.log);
+		const github = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_BRANCH_EVENT, false, jiraClient.baseURL) ? gitHubClient : context.github;
+		const jiraPayload = await transformBranch(github, webhookPayload);
 
 		if (!jiraPayload) {
 			context.log(
-				{ noop: "no_jira_payload_create_branch" },
+				{ noop: "no_jira_payload_create_branch" },
```
Comment (mboudreau) [inline src/github/branch.ts:35]: /nit `noop` data is from oldschool code that needs to be removed.  If you're editing anything else in this PR, please remove it as it adds no value.

File: src/github/branch.ts [hunk 5]
```diff
@@ -104,17 +99,12 @@ export const deleteBranch = async (context: CustomContext, jiraClient): Promise<
...
 	const issueKeys = issueKeyParser().parse(payload.ref);
 
-	if (_.isEmpty(issueKeys)) {
-		context.log(
-			{ noop: "no_issue_keys" },
-			"Halting further execution for deleteBranch since issueKeys is empty"
-		);
-		return undefined;
+	if (isEmpty(issueKeys)) {
+		context.log({ noop: "no_issue_keys" }, "Halting further execution for deleteBranch since issueKeys is empty");
```
Comment (mboudreau) [inline src/github/branch.ts:103]: /nit same here.

File: src/sync/commits.ts (hunk 1 above)
Comment (mboudreau) [reply src/sync/commits.ts:32]: to make sure this never happens again, please add a specific return type for the function.

Comment (joshkay10) [reply src/sync/commits.ts:38]: typed

**Review by mboudreau — APPROVED** [review]

**Pull request merged**